File: 2021/java/1480.running-sum-in-1d-array.java
class Solution {
    // presum, sum[i] = nums[i] + sum[i-1]
    public int[] runningSum(int[] nums) {
        if (nums == null || nums.length == 0) return new int[0];
        int len = nums.length;
        int[] res = new int[len];
        res[0] = nums[0];
        for (int i = 1; i < len; i++) {
            res[i] = nums[i] + res[i - 1];
        }
        return res;
    }
}

File: 2021/python/1480.running-sum-in-1d-array.py
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        if not nums or len(nums) == 0:
            return nums
        res = []
        res.append(nums[0])
        for i in range(1, len(nums)):
            res.append(res[-1] + nums[i])
        return res
